src/main/java/edu/umich/ctools/sectionsUtilityTool/Friend.java

```diff
@@ -0,0 +1,362 @@
+			sslInitialized = false;
+		}
+		else
+		{	
+			// key store
+			systemProps.put("javax.net.ssl.keyStoreType", KEYSTORETYPE_PKCS12);
+			systemProps.put("javax.net.ssl.trustStoreType", TRUSTSTORETYPE_JKS);
+
+			if (ksFileName.isEmpty())
+			{
+				// log error for missing keystore file path
+				M_log.error(this + " init the ctools.friend.keystorefile path is not defined. ");
+			}
+			else
+			{
+				systemProps.put("javax.net.ssl.keyStore",ksFileName);
+			}
+
+			if (ksPwd.isEmpty())
+			{
+				// log error for missing keystore password
+				M_log.error(this + " init the ctools.friend.keystorepassword is not defined. ");
+			}
+			else
+			{
+				systemProps.put("javax.net.ssl.keyStorePassword",ksPwd);
+			}
+
+			// set system properties
+			System.setProperties(systemProps);
+
+			sslInitialized = true;
+		}
+	}	
+
+	/**
+	 * Returns to uninitialized state.
+	 */
+	public void destroy()
+	{
+		M_log.info(this +".destroy()");
+
+		if (sslInitialized)
+		{
+			// remove ssl system settings
+			// configure JSSE system properties: 
+			// http://fusesource.com/docs/broker/5.3/security/SSL-SysProps.html
+			Properties systemProps = System.getProperties();
+			//systemProps.remove("javax.net.debug");
+			//important: http://stackoverflow.com/questions/6680416/apache-cxf-exception-in-ssl-communication-sockettimeout
+			//java.lang.System.setProperty("sun.security.ssl.allowUnsafeRenegotiation", "true");
+			systemProps.remove("javax.net.ssl.keyStoreType");
+			systemProps.remove("javax.net.ssl.trustStoreType");
+			systemProps.remove("javax.net.ssl.keyStore");
+			systemProps.remove("javax.net.ssl.keyStorePassword");
+			System.setProperties(systemProps);
+
+			sslInitialized = false;
+		}
+	}	
+
+	/*
+	 * Check if an account exists
+	 * Return a single integer with the following (It all succeeds of fails)
+	 * @param email friend account Email
+	 * @return
+	 * -1 = bad e-mail address
+	 * 0 = no friend account
+	 * 1 = friend account exists 
+	 *
+	 */
+	public CheckAccountExistsResponse checkAccountExist(String email) {
+		M_log.debug("Friend checkAccountExists() called");
+		CheckAccountExistsResponse response = CheckAccountExistsResponse.FRIEND_ACCOUNT_DOES_NOT_EXIST;
+		int rv = 0;	// default to be "no friend account"
+		try {
+			
+			Object[] params = new Object[]{new String[] {email}};
+			Object[] results = (Object[]) Xclient.execute(DO_ACCOUNT_EXIST_WS, params);
+			// though the friend XML-RPC service supports batch call mode, 
+			// due to the ctools event model ( one "user added event" per user), 
+			// we will pass only one email address to the doAccountsExit call for now
+			// hence we should only get one call result returned
+			if (results != null && results.length == 1)
+			{
+				rv = ((Integer) results[0]).intValue();
+			}
+		}
+		catch (Exception e) {
+			M_log.warn("Friend checkAccountExist(): email address " + email + " " + e.getMessage());
```

Comment:
If there is an exception this will log but then keep going and return response.  Is that reasonable?